src/Export_Command.php

```diff
@@ -207,9 +207,34 @@ static function ( $file_path ) {
 
 	private static function get_filename_template( $filename_format ) {
 		$sitename = sanitize_key( get_bloginfo( 'name' ) );
+
 		if ( empty( $sitename ) ) {
 			$sitename = 'site';
 		}
+
+		/**
+		 * Get the potential filename.
+		 */
+		$potential_file_name = str_replace( [ '{site}', '{date}', '{n}' ], [ $sitename, date( 'Y-m-d' ), '%03d' ], $filename_format ); // phpcs:ignore WordPress.DateTime.RestrictedFunctions.date_date
+
+		/**
+		 * Get the length of the potential file name.
+		 */
+		$potential_file_name_size = strlen( $potential_file_name );
+
+		/**
+		 * If the potential filename size exceeds 99 bytes, then remove the extra bytes.
+		 */
+		if ( $potential_file_name_size > 99 ) {
+			$extra_bytes   = $potential_file_name_size - 99;
+			$sitename_size = strlen( $sitename );
+
+			/**
+			 * Remove the extra bytes from the sitename.
+			 */
+			$sitename = substr( $sitename, 0, ( $sitename_size - $extra_bytes ) );
```

Comment:
Using `strlen` and `substr` on multibyte strings will lead to corrupted characters, so I assume this will currently not work for titles with multibyte characters in the wrong position.

Reply:
This should use the multibyte versions of the string operations if they are available.